controllers/storagecluster/storagecluster_controller.go

```diff
@@ -131,6 +132,7 @@ func (r *StorageClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	enqueueStorageClusterRequest := handler.EnqueueRequestsFromMapFunc(
 		func(context context.Context, obj client.Object) []reconcile.Request {
 			// Get the StorageCluster objects
+			r.ctx = ctrllog.IntoContext(context, r.Log)
```

Comment:
What are the benefits of adding the logger to the context? If you want to control the logger of each reconcile iteration there is a controller option to provide a function to create a logger for each processed event and store it on context sent to the reconcile call for that event.

Reply:
I've had this particular patch for a very, very long time. It looks like the issue I was running into (there was a `nil` pointer reference to `r.ctx` somewhere, I don't remember now) was resolved in a different place by PR #1748 . I can probably safely remove it.

Reply:
@jarrpa I think you are referring to https://github.com/red-hat-storage/ocs-operator/pull/2232

Reply:
Sure seems so! I'll remove this, then.